src/test/java/bookface/logic/commands/LoanCommandTest.java

```diff
@@ -0,0 +1,121 @@
+package bookface.logic.commands;
+
+import static bookface.logic.commands.LoanCommand.ALREADY_LOANED;
+import static bookface.testutil.TypicalDates.TYPICAL_DATE;
+import static bookface.testutil.TypicalPersons.getAllTypicalBookFaceData;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import bookface.commons.core.Messages;
+import bookface.commons.core.index.Index;
+import bookface.model.Model;
+import bookface.model.ModelManager;
+import bookface.model.UserPrefs;
+import bookface.model.book.Book;
+import bookface.model.person.Person;
+import bookface.testutil.TypicalIndexes;
+
+public class LoanCommandTest {
+    private final Model model = new ModelManager(getAllTypicalBookFaceData(), new UserPrefs());
+
+    @Test
+    public void execute_loanedBook_throwsCommandException() {
+        Book bookToLoan = model.getFilteredBookList().get(TypicalIndexes.INDEX_FIRST_BOOK.getZeroBased());
+        Person personToLoan = model.getFilteredPersonList().get(TypicalIndexes.INDEX_FIRST_PERSON.getZeroBased());
+        LoanCommand loanCommand = new LoanCommand(TypicalIndexes.INDEX_FIRST_PERSON, TypicalIndexes.INDEX_FIRST_BOOK);
+        String expectedMessage = String.format(ALREADY_LOANED);
+        ModelManager expectedModel = new ModelManager(model.getBookFace(), new UserPrefs());
+        expectedModel.loan(personToLoan, bookToLoan, TYPICAL_DATE);
+        CommandTestUtil.assertCommandFailure(loanCommand, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredPersonList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        LoanCommand loanCommand = new LoanCommand(outOfBoundIndex, TypicalIndexes.INDEX_FIRST_BOOK);
+        CommandTestUtil.assertCommandFailure(loanCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+
+    // For this, it shows "Book is already loaned out." instead of MESSAGE_LOAN_SUCCESS, but I'm not sure why this
```

Comment:
Is this fixed?

Reply:
For the tests, sorry I'm still unable to fix them. But can pin the blame on me if asked who's in charge of loan and return tests so I dont think team will be penalized for this.

Reply:
Oh I guess we can merge first then - we can figure this out later tonight or tomorrow during the meeting